pydatalab/src/pydatalab/models/traits/__init__.py

```diff
@@ -0,0 +1,180 @@
+from typing import Any
+
+from pydantic import (
+    BaseModel,
+    Field,
+    model_validator,
+)
+
+from pydatalab.models.people import Person
+from pydatalab.models.utils import Constituent, PyObjectId
+
+
+class HasOwner(BaseModel):
+    creator_ids: list[PyObjectId] = Field(
+        default_factory=list, description="The database IDs of the user(s) who created the item."
+    )
+
+    creators: list[Person] | None = Field(
+        None, description="Inlined info for the people associated with this item."
+    )
+
+
+class HasRevisionControl(BaseModel):
+    revision: int = Field(1, description="The revision number of the entry.")
+
+    revisions: dict[int, Any] | None = Field(
+        None,
+        description="An optional mapping from old revision numbers to the model state at that revision.",
+    )
+
+
+class HasBlocks(BaseModel):
+    blocks_obj: dict[str, Any] = Field(
+        default_factory=dict, description="A mapping from block ID to block data."
+    )
+
+    display_order: list[str] = Field(
+        default_factory=list, description="The order in which to display block data in the UI."
+    )
+
+
+class HasSynthesisInfo(BaseModel):
+    """Trait mixin for models that have synthesis information."""
+
+    synthesis_constituents: list[Constituent] = Field(
+        default_factory=list,
+        description="A list of references to constituent materials giving the amount and relevant inlined details of consituent items.",
+    )
+
+    synthesis_description: str | None = Field(
+        None, description="Free-text details of the procedure applied to synthesise the sample"
+    )
+
+    @model_validator(mode="before")
+    @classmethod
+    def add_missing_synthesis_relationships(cls, values):
+        """Add any missing sample synthesis constituents to parent relationships"""
```

Comment:
I've moved this whole module, but want to check this validator as it seems a bit excessive